pkg/controller/provider/container/openstack/client.go

```diff
@@ -126,15 +126,20 @@ func (r *Client) Authenticate() (err error) {
 		} else {
 			cacert := []byte(r.getStringFromSecret(CACert))
 			roots := x509.NewCertPool()
-			ok := roots.AppendCertsFromPEM(cacert)
-			if !ok {
-				r.Log.Info("the CA certificate is malformed or was not provided, falling back to system CA cert pool")
+
+			if len(cacert) == 0 {
```

Comment:
From https://pkg.go.dev/crypto/tls#Config:
>  // RootCAs defines the set of root certificate authorities
>	// that clients use when verifying server certificates.
>	// If RootCAs is nil, TLS uses the host's root CA set.
>	RootCAs *[x509](https://pkg.go.dev/crypto/x509).[CertPool](https://pkg.go.dev/crypto/x509#CertPool)

So maybe we can simplify this block.